mmv1/third_party/terraform/services/sql/resource_sql_database_instance.go.tmpl

```diff
@@ -1434,6 +1435,61 @@ func stringContainsSlice(arr []string, str string) bool {
 	return false
 }
 
+// enhancedBackupManagerCustomizeDiff suppresses diff changes to settings.backup_configuration
+// when the SQL instance is managed by Google Cloud Backup and Disaster Recovery (DR) Service.
+func enhancedBackupManagerCustomizeDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{}) error {
+	project := ""
+	if v := d.Get("project"); v != nil {
+		project = v.(string)
+	}
+
+	name := ""
+	if v := d.Get("name"); v != nil {
+		name = v.(string)
+	}
+
+	if project == "" || name == "" {
+		return nil
+	}
+
+	config := meta.(*transport_tpg.Config)
+	client := config.NewSqlAdminClient(config.UserAgent)
+	instance, err := client.Instances.Get(project, name).Do()
+    if err != nil {
+        // If we can't read the remote instance, don't fail the plan; just skip suppression.
+        var gerr *googleapi.Error
+        if errors.As(err, &gerr) {
+            return nil
+        }
+        return nil
+    }
+
+    if instance == nil || instance.Settings == nil || instance.Settings.BackupConfiguration == nil {
+        return nil
+    }
+
+    // Only suppress when backend reports ENHANCED backup manager.
+    if !strings.EqualFold(instance.Settings.BackupConfiguration.BackupTier, "ENHANCED") {
+        return nil
+    }
+
+	// Clear the diff since ENHANCED backup manager is enabled
+	if d.HasChange("settings.0.backup_configuration") {
```

Comment:
I'm fairly certain that this won't have the impact you want - it will clear the diff at plan time, but not prevent the fields from being updated if an update runs. It's possible I'm wrong but it would be good to check.